common/src/tracer.rs

```diff
@@ -159,6 +161,12 @@ impl OtelTracer {
     pub fn set_span_parent_to(span: &Span, parent_span: &Span) {
         span.set_parent(parent_span.context());
     }
+
+    pub fn kill_tracer_on_err(e: impl Debug + Display) -> impl Debug + Display {
```

Comment:
This feels quite clunky and I am not a fan of the way this function is used in error handling.

Why is it actually needed? Is it a result of the fact panics are used for some error handing in initialisation?

Reply:
Yes, if the program panics then spans or error/tracing messages may not be emitted. This seemed the most elegant solution, an alternative however would be handling those initialisation panics in code.

Reply:
If it is specifically panicking that is the issue then I would prefer not doing this in favour of https://github.com/STFC-ICD-Research-and-Design/supermusr-data-pipeline/issues/161 (where the only reason for a panic would be a programming error that would realistically be an issue anyway).

Reply:
Ah of course, that would cause the tracer to shut down, eliminating the need for `OtelTracer::kill_tracer_on_err` would it not?